### DISCUSSION TITLE
Action / Reward functions?

### DISCUSSION THREAD

**Discussion by kristopher-wood** [online-ml/river#682] [General]
Has anyone implemented or even given any thought to how one would implement action and reward functions in online learning similar to what is done in reinforcement learning? From what I've read, there is no reason the same math used in, for example, actor critic models, shouldn't be able to be used online rather than batched. I'm more curious as to whether anyone's done it and what your thought are on the possibilities in an online setting.

What do you think?

**Comment by MaxHalford** [comment]
We have really delved into reinforcement learning. It's such a large topic and there are many other people who know much more about it than we do. That being said, I wouldn't be against a reinforcement learning module for small tasks at some point. I think we're just waiting for a use case and/or someone with the necessary expertise who is also willing to contribute. Hope that makes sense.

By the way, note that we have @etiennekintzler who contributed [`EpsilonGreedyRegressor`](https://riverml.xyz/latest/api/expert/EpsilonGreedyRegressor/) and [`UCBRegressor`](https://riverml.xyz/latest/api/expert/UCBRegressor/), which are bandits, which are a special case of reinforcement learning.
